harmony_regridding_service/grid.py

```diff
@@ -179,12 +179,72 @@ def get_geographic_area_extent(
 ) -> tuple[float, float, float, float]:
     """Return the geographic area extent.
 
-    Compute the latitude and longitude for every grid cell in the
-    projected_area's grid and return the area extent in lon and lat.
+    Compute the latitude and longitude for the center and each corner for every
+    grid cell in the projected_area's grid and return the area extent in lon
+    and lat.
 
     """
-    lons, lats = projected_area.get_lonlats()
-    return (np.min(lons), np.min(lats), np.max(lons), np.max(lats))
+    target_crs = CRS(4326)
+    tf = Transformer.from_crs(projected_area.crs, target_crs, always_xy=True)
+
+    # Get the projected 1D coordinates for the input grid.
+    center_x_1d_coords, center_y_1d_coords = projected_area.get_proj_vectors()
+
+    # get the right and left sides of each cell
+    right_x_1d_coords = center_x_1d_coords - projected_area.resolution[0] / 2.0
+    left_x_1d_coords = center_x_1d_coords + projected_area.resolution[0] / 2.0
+
+    # get the top and bottom sides of each cell
+    top_y_1d_coords = center_y_1d_coords + projected_area.resolution[1] / 2.0
+    bottom_y_1d_coords = center_y_1d_coords - projected_area.resolution[1] / 2.0
+
+    # Use only the unique values from the right, center, left values, since the
+    # right of cell 1 is the same as the left of cell2 this reduces the total
+    # number of points we need to transform.
+    round_to = 6
+    x_coords = np.unique(
+        (
+            np.round(right_x_1d_coords, round_to),
+            np.round(center_x_1d_coords, round_to),
+            np.round(left_x_1d_coords, round_to),
+        )
+    )
+
+    # Use only the unique values from the top, center, bottom y values
+    y_coords = np.unique(
+        (
+            np.round(top_y_1d_coords, round_to),
+            np.round(center_y_1d_coords, round_to),
+            np.round(bottom_y_1d_coords, round_to),
+        )
+    )
+
+    # Process the x and y coordinates in chunks (max of 4000000 points at a time.)
+    coord_chunk_size = 2000
+    n_x_chunks = len(x_coords) // coord_chunk_size + 1
+    n_y_chunks = len(y_coords) // coord_chunk_size + 1
+
+    lon_min = np.inf
+    lat_min = np.inf
+    lon_max = -np.inf
+    lat_max = -np.inf
+
+    for x_coord_chunk in np.array_split(x_coords, n_x_chunks):
+        for y_coord_chunk in np.array_split(y_coords, n_y_chunks):
+            x_chunk_grid, y_chunk_grid = np.meshgrid(x_coord_chunk, y_coord_chunk)
+            chunk_lons, chunk_lats = tf.transform(x_chunk_grid, y_chunk_grid)
+            lon_min = np.nanmin([lon_min, np.nanmin(chunk_lons)])
+            lat_min = np.nanmin([lat_min, np.nanmin(chunk_lats)])
+            lon_max = np.nanmax([lon_max, np.nanmax(chunk_lons)])
+            lat_max = np.nanmax([lat_max, np.nanmax(chunk_lats)])
```

Comment:
I think this makes sense now and should keep a handle on number of points being held in memory for large grids. Nice!